### PR TITLE
fix(search): update to only link title in search results

### PR DESCRIPTION
**Pull request by jsundquist**
## Hey, I just made a Pull Request!

This change updates the way that the search page displays links.  When a user opens the search modal window, the entire block of text is a link.  This change makes it so that only the primary text (header) is a link. This aligns with how many web sites work and makes it more readable for users when hovering over the text.

Previous:
previous version with link for the entire block

Updated:
Updated version with link for only the header

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by kuangp — COMMENTED** [review]
We probably want to propagate this change to the other `*SearchResultListItem` components as well. Here are the ones I'm aware of: 
- https://github.com/backstage/backstage/blob/master/plugins/adr/src/search/AdrSearchResultListItem.tsx
- https://github.com/backstage/backstage/blob/master/plugins/search-react/src/components/DefaultResultListItem/DefaultResultListItem.tsx
- https://github.com/backstage/backstage/blob/master/plugins/stack-overflow/src/search/StackOverflowSearchResultListItem/StackOverflowSearchResultListItem.tsx
- https://github.com/backstage/backstage/blob/master/plugins/techdocs/src/search/components/TechDocsSearchResultListItem.tsx

**Comment by jsundquist** [comment]
Thanks for pointing those out @kuangp . Will look into updating those as well so its consistent across the board.

**Review by nuritizra — DISMISSED** [review]
thanks, @emmaindal what do you think about this? is this intended that the entire item should be clickable.

File: plugins/catalog/src/components/CatalogSearchResultListItem/CatalogSearchResultListItem.tsx [hunk 1]
```diff
@@ -80,13 +80,17 @@ export function CatalogSearchResultListItem(
             primaryTypographyProps={{ variant: 'h6' }}
             primary={
               props.highlight?.fields.title ? (
-                <HighlightedSearchResultText
-                  text={props.highlight.fields.title}
-                  preTag={props.highlight.preTag}
-                  postTag={props.highlight.postTag}
-                />
+                <Link noTrack to={result.location} onClick={handleClick}>
```
[the remark this answers is not in the document]
Comment (jsundquist) [reply plugins/catalog/src/components/CatalogSearchResultListItem/CatalogSearchResultListItem.tsx]: Updated so all of the link tags are outside of the ternary now.

**Comment by jsundquist** [comment]
@nuritizra this is ready to re-review. Let me know if there is any additional items.

**Review by emmaindal — APPROVED** [review]
Looks great to me, will let @nuritizra and @camilaibs to have a look again before merging!

**Review by camilaibs — APPROVED** [review]
LGTM

File: plugins/adr/src/search/AdrSearchResultListItem.tsx [hunk 2]
```diff
@@ -64,20 +64,24 @@ export function AdrSearchResultListItem(props: {
   };
 
   return (
-    <Link noTrack to={result.location} onClick={handleClick}>
+    <>
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/adr/src/search/AdrSearchResultListItem.tsx:67]: Super optional, we can do this in a separate pull request 🙂

**Review by Rugvip — APPROVED** [review]
Thank you, looks good! 👍 

Gonna :shipit:

**Pull request merged**